homeassistant/helpers/typing.py

```diff
@@ -14,6 +14,7 @@
     dir_with_deprecated_constants,
 )
 
+type AsyncCallable[**_P, _R_co] = Callable[_P, Coroutine[Any, Any, _R_co]]
```

Comment:
Note: as search for `Callable\[.*, Coroutine\[.*\]\]` in `homeassistant/components` returns 142 results in 109 files.
I've adjusted here 6 components (`homeassistant/components/a*`) to show how it would be used.